WIP implement mentions + findUser falls back to webfinger

traumschule opened the pull request:
see #26, #27

1. [X] match @user@domain or @user (base_url is used)
1. [X] wrap in html
1. [X] create mention object
1. [X] add cc
1. [ ] test search with non-pgi

Note: the output specification requires `"cc": [<user_id>]`. When mentioning multiple users backend raised 500 because the cc list was inserted as an array. Hence line 46 in *Publish.svelte* is `ap_object.cc = mentions.map(m => m.href);` instead of `ap_object.cc = [mentions.map(m => m.href)];`

Commit:
implement mentions (fixes #27)

autogestion commented on philip/src/components/Publish.svelte:
```diff
@@ -27,14 +39,17 @@
-    const data = wrapHashTagsWithLink(wrapLinksWithTags(content));
-
-    let ap_object = getCreateObject(data, tags);
+    const data = wrapMentions(wrapHashTagsWithLink(wrapLinksWithTags(content)));
```
:)))

autogestion commented:
![Screenshot_39](https://user-images.githubusercontent.com/1098257/77260819-26c1ba00-6c82-11ea-8267-3489ab9006c8.png)
tested mentions. looks like domain part should be cropped. Or maybe put the hole username (with domain) inside `<span>$2</span>`

autogestion commented:
and renders incorrect url in cc (which will be used by server to notify person which was mentioned)
![Screenshot_41](https://user-images.githubusercontent.com/1098257/77261327-a1d89f80-6c85-11ea-8b19-e3cc97b86a28.png)

Commit:
Search: fallback to webfinger (fixes #26)

- Publish ("Create"): separate publish and sendPost 
- Search: factor out requests to utils/user.js
- needs testing for non-pgi

traumschule commented:
thanks for testing! needs some more :)

autogestion commented:
Have tested mentions, works fine
![Screenshot_42](https://user-images.githubusercontent.com/1098257/77318193-8fea1180-6d04-11ea-8ec0-66fdd1eaed6d.png)
Mentioned user do not follow test acc, still got post where he is mentioned


Same time, fallback to webginger failed when tried to search for `putinhujlo@pleroma.site`
